src/reducers/search.js

```diff
@@ -79,14 +82,13 @@ const reducer = createReducer()
     },
   }))
 
-  .when(types.SORT_SEARCH_RESULTS_BY, (state, { payload }) => ({
-    ...state,
-    query: {
-      ...state.query,
-      page: initialState.query.page,
-      sortBy: payload,
-    },
-  }))
+  .when(types.SORT_SEARCH_RESULTS_BY, (state, { payload }) => {
+    const newState = Object.assign({}, state);
+    newState.query.sortBy.key = payload.key;
+    newState.query.sortBy.order = payload.order;
+    newState.query.page = 1;
+    return newState;
+  })
```

Comment:
It would be more clear and easy to use destruction syntax to implement this merging.